### PR TITLE
tests: Support selecting multiple cases to run

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

We can't select more than one integration test case to run.

### What is changed and how it works?

Support selecting multiple cases to run.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -127,12 +127,12 @@ if [ "${1-}" = '--debug' ]; then
...
 
-# set to the case name you want to run only for debug
-do_case=""
+# List the case names to run, separate them by space, eg. "binlog gencol"
+do_cases=""
 
 for script in ./*/run.sh; do
     test_name="$(basename "$(dirname "$script")")"
-    if [[ $do_case != "" && $test_name != $do_case ]]; then
+    if [[ $do_cases != "" && ! $do_cases =~ (^|[[:space:]])$test_name($|[[:space:]]) ]]; then
```
Comment (amyangfei) [inline tests/run.sh]: note `[[` is not available in `sh`, and bash is superset of sh, it is better to change the shebang to `#!/bin/bash`

File: tests/run.sh [hunk 2]
```diff
@@ -127,12 +127,12 @@ if [ "${1-}" = '--debug' ]; then
     read line
 fi
 
-# set to the case name you want to run only for debug
-do_case=""
+# List the case names to run, separate them by space, eg. "binlog gencol"
+do_cases=""
```
Comment (WangXiangUSTC) [inline tests/run.sh]: how about use array in shell  https://stackoverflow.com/questions/1878882/arrays-in-unix-shell

Comment (suzaku) [reply tests/run.sh]: Is there an easy way to test if a case is among the selected ones with array?

Comment (WangXiangUSTC) [reply tests/run.sh]: if do_cases' length is not 0, just run code like:
```
for case in $do_cases; do
      sh tests/$case/run.sh
done
```

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by suzaku** [comment]
@amyangfei PTAL

**Comment by GregoryIan** [comment]
@amyangfei PTAL

**Pull request merged**